lib/Backend/Vault/Kv.hs

```diff
@@ -224,6 +224,17 @@ kvReadSecret backend path = do
   where
     readSecret env = (I.routes env ^. I.readSecret) (vbMount backend) (vbToken backend)
 
+    keyAndValueToField :: CofferSpecials -> (Text, Text) -> Either T.Text (FieldKey, Field)
+    keyAndValueToField cofferSpecials (key, value) = do
+      _modTime <- maybeToEither ("No field 'dateModified' at key '" <> key <> "'") $ cofferSpecials ^? fields . at key . _Just . dateModified
```

Comment:
Done

Reply:
I don't think this bit is done yet.

My point was that saying "No field 'dateModified' at key `k`" isn't correct. It's not that the field `dateModified` wasn't found - it's the key itself that wasn't found.

So saying `Could not find coffer metadata for field '<key>' at '<path>'` is more correct.

Reply:
Oops sorry I missed your last fixup!

Reply:
```hs
        mkError :: Text -> Builder
        mkError field =
          "Could not find coffer metadata for field '" +| field
            |+ "' at key '" +| key
            |+ "' at '" +| path |+ "'"
```

Ok there's a bit of terminology mixup here.

* `key` here represents what we call "field key" or "field name" in other places of the codebase, e.g. "username", "password"
* `field` here represents _a piece of metadata_ for a particular "field", e.g. the date this field was last modified.

I suggest:
1. renaming the variable `key` to `fieldKey` to make this more clear
2. We shouldn't say ...
    ```
    Could not find coffer metadata for field 'dateModified' at key 'password' at ...
    ```
    ... because 'dateModified' didn't fail at all. What we failed to find was the entire `FieldMetadata` object for the field `password`. Instead, we should say:
    ```
    Could not find coffer metadata for field 'password' at ...
    ```
3. Looking up a `FieldMetadata` for a given "entry field" (`cofferSpecials ^? fields . at key`) and throwing an exception if it's not found doesn't need to be done twice. We should do it just once.

Reply:
Would `fieldName` be clearer still.

Reply:
@treeowl yes indeed! I've actually been meaning to open an issue to unify some names.

The [design doc](https://www.notion.so/serokell/Password-sharing-2-232733503bb4469d85713d08f3b3a5e4#2459058a49df42ed84bcc37dd1d6fd4c) uses "field name" and "field contents". The `CLI` module followed this naming convention. Some of the core modules were developed in parallel and use "field key" and "field value" respectively. So there's a disconnect there.

I'll write this issue now before I forget again :sweat_smile:

Reply:
Done: #71